components/fxa-client/src/profile.rs

```diff
@@ -40,13 +40,23 @@ impl FirefoxAccount {
                 }
                 Ok(response_and_etag.response)
             }
-            None => match self.profile_cache {
-                Some(ref cached_profile) => Ok(cached_profile.response.clone()),
-                None => Err(ErrorKind::UnrecoverableServerError(
-                    "Got a 304 without having sent an eTag.",
-                )
-                .into()),
-            },
+            None => {
+                match self.profile_cache.take() {
+                    Some(ref cached_profile) => {
+                        // Update `cached_at` timestamp.
```

Comment:
@eoger is it reasonable to update the `cached_at` time here based on `304 Not Modified` from the server?

Reply:
Sure I'm not sure what this gives us but ok!

Reply:
IIUC, previously we would cache the data in memory for 2 minutes and then once the 2 minutes has expired, every call to `get_profile` will hit the network to check freshness of the etag. With this chance, we'll do the etag check only once every 2 minutes, potentially avoiding some calls to the network if `get_profile` is called several times in quick succession.